src/utils/editor/index.js

```diff
@@ -74,6 +74,55 @@ function clearLineClass(codeMirror, className) {
   });
 }
 
+function isTextForSource(sourceText) {
+  return !sourceText.get("loading") && !sourceText.get("error");
+}
+
+function breakpointAtLine(breakpoints, line) {
+  return breakpoints.find(b => {
+    return b.location.line === line + 1;
+  });
+}
+
+function getTextForLine(codeMirror, line) {
+  return codeMirror.getLine(line - 1).trim();
+}
+
+function getCursorLine(codeMirror) {
+  return codeMirror.getCursor().line;
+}
+/**
+ * Forces the breakpoint gutter to be the same size as the line
+ * numbers gutter. Editor CSS will absolutely position the gutter
+ * beneath the line numbers. This makes it easy to be flexible with
+ * how we overlay breakpoints.
+ */
+function resizeBreakpointGutter(editor) {
+  const gutters = editor.display.gutters;
+  const lineNumbers = gutters.querySelector(".CodeMirror-linenumbers");
+  const breakpoints = gutters.querySelector(".breakpoints");
+  breakpoints.style.width = `${lineNumbers.clientWidth}px`;
+}
+
+function traverseResults(e, ctx, query, dir, modifiers) {
+  e.stopPropagation();
+  e.preventDefault();
+  if (dir == "prev") {
+    findPrev(ctx, query, true, modifiers);
+  } else if (dir == "next") {
+    findNext(ctx, query, true, modifiers);
+  }
+}
+
+function onMouseUp(ctx, modifiers) {
+  const query = ctx.cm.getSelection();
+  if (ctx.cm.somethingSelected()) {
+    find(ctx, query, true, modifiers);
+  } else {
+    removeOverlay(ctx);
+  }
+}
```

Comment:
hmm, i think this could be moved into the Editor component as we might want to change the Editor mouse up behavior